From 0fc7201bd45a58cfb354dc0d52cacae27b49ecbb Mon Sep 17 00:00:00 2001
From: Torge Matthies <tmatthies@codeweavers.com>
Date: Mon, 24 Oct 2022 19:30:25 +0200
Subject: [PATCH 2/4] ntdll: Add 5-byte nop to start of
 KiUserCallbackDispatcher.

Overwatch 2 hooks KiUserCallbackDispatcher by overwriting the first five bytes with a jump, and returning to
just after the jump. Make sure there is a five-byte instruction for it to replace.

Signed-off-by: Torge Matthies <openglfreak@googlemail.com>
---
 dlls/ntdll/signal_x86_64.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/ntdll/signal_x86_64.c b/dlls/ntdll/signal_x86_64.c
index 11111111111..11111111111 100644
--- a/dlls/ntdll/signal_x86_64.c
+++ b/dlls/ntdll/signal_x86_64.c
@@ -700,6 +700,7 @@ void WINAPI user_callback_dispatcher( ULONG id, void *args, ULONG len )
  * FIXME: not binary compatible
  */
 __ASM_GLOBAL_FUNC( KiUserCallbackDispatcher,
+                  ".byte 0x0f, 0x1f, 0x44, 0x00, 0x00\n\t" /* Overwatch 2 replaces the first 5 bytes with a jump */
                   "andq $0xFFFFFFFFFFFFFFF0, %rsp\n\t"
                   __ASM_SEH(".seh_endprologue\n\t")
                   "call " __ASM_NAME("user_callback_dispatcher") "\n\t"
-- 
2.39.0

